core/src/it/scala/org/ensime/core/RefactoringHandlerSpec.scala

```diff
@@ -268,27 +224,31 @@ class RefactoringHandlerSpec extends EnsimeSpec
     val analyzer = analyzerRef.underlyingActor
 
     val procId = 1
-    analyzer.handleRefactorPrepareRequest(
-      new PrepareRefactorReq(
-        procId, 'rename, RenameRefactorDesc("doItNow", new File(file.path), 43, 47), false
+    val result = analyzer.handleRefactorRequest(
+      new RefactorReq(
+        procId, RenameRefactorDesc("doItNow", new File(file.path), 43, 47), false
       )
     )
-    analyzer.handleRefactorExec(
-      new ExecRefactorReq(procId, RefactorType.Rename)
-    )
-    val formatted = readSrcFile(file, encoding)
-    val expectedContents = contents(
-      "package org.ensime.testing",
-      "trait Foo {",
-      "def doItNow(",
-      ") = \"\"",
-      "}",
-      ""
-    )
-    formatted should ===(expectedContents)
+    val diffFile = result match {
+      case RefactorDiffEffect(_, _, f) => f.canon
+      case _ => fail()
+    }
+
+    val diffContents = diffFile.readString()(analyzer.charset)
+
+    val relevantExpectedPart = s"""|@@ -2,3 +2,3 @@
+                                   | trait Foo {
+                                   |-def doIt(
+                                   |+def doItNow(
+                                   | ) = ""
+                                   |""".stripMargin
+    val expectedContents = expectedDiffContent(file.path, relevantExpectedPart)
+
+    diffContents should ===(expectedContents)
+    diffFile.delete()
```

Comment:
these `.delete()` calls should be in a `finally` block or they'll not be cleaned up when the tests fail.